src/Headers.php

```diff
@@ -58,58 +58,49 @@ class Headers implements Countable, Iterator
      * will be lazy loaded)
      *
      * @param  string $string
-     * @param  string $EOL EOL string; defaults to {@link EOL}
+     * @param  string $EOL DEPRECATED
      * @throws Exception\RuntimeException
      * @return Headers
      */
-    public static function fromString($string, $EOL = self::EOL)
+    public static function fromString($string, $EOL = null)
     {
         $headers     = new static();
         $currentLine = '';
         $emptyLine   = 0;
 
         // iterate the header lines, some might be continuations
-        $lines = explode($EOL, $string);
-        $total = count($lines);
-        for ($i = 0; $i < $total; $i += 1) {
-            $line = $lines[$i];
-
+        foreach (preg_split('/\R/', $string) as $line) {
```

Comment:
Mail with mixed line do exists in the wild, and even if it's not rfc compliant, i do want to handle them

for now i fail to find an exemple string that would made us vulnerable
(if you have an exemple in mind ...)